### ISSUE TITLE
Some questions regarding implementation 

### ISSUE THREAD

**Issue by maggiemae13** [DeepMReye/DeepMReye#83]
Hello!

With the documentation still down, I have a few questions regarding implementing DeepMReye on some data. I've looked through the paper a bit as well, but just need some more clarification. 

1. Preprocessing choices:
  - What steps are necessary for a "minimal" pipeline. I assume aligning the scans to one another, but is there anything else that absolutely needs to be done in order for an accurate model (gradient nonlinearity correction, distortion correction, motion, etc)? The paper notes a 3-step co-registration, is this replicated entirely (including gaze centering) in the preprocess.run_participant command, or is there something else we need to do/add to ensure everything is aligned/centered correctly?
  - Conversely, are there any steps that can interfere with, degrade model performance or need to be considered carefully? I.E. sometimes we interpolate our timeseries to a faster TR when doing slice timing correction, should this be skipped for data we want to use to train and/or predict? I read the information on **Subimaging temporal resolution in the** paper, but it seems to just be including more gaze labels per TR rather than reslicing the timeseries to a different temporal resolution.

2. Training/Fine-tuning
  - Are there recommendations regarding training the model from scratch vs fine-tuning the pre-trained weights?
  - Is it better to train/fine-tune on the dataset as a whole (i.e. across all subjects) or in a subject-specific manner (provided there is enough data per subject)?
  - Does it matter if the gaze coordinates are in pixel or DVA space (other than for visualization purposes)?

Thanks for the help! Also, if these questions are addressed in the documentation text and you still have it available (even if in html/css/xml format), I'm happy to just comb through it myself. Please feel free to send it to maggiemaemell@gmail.com, if that's easier!

**Comment by CYHSM** [comment]
Hey @maggiemae13 thanks for the questions, hope the below can help you. We hope we can get a version of the docs up and running again soon.

1.
- The co-registration step "just" transforms your scans to MNI space, so you must ensure you run all other preprocessing steps before (motion correction etc...)
- We have not yet quantified how interpolating to a faster TR impacts the performance of the model but I would suggest retraining the model in that case instead of using the provided weights

2.
- We have not really quantified this either. I suspect if your viewing prior matches one of the pre-trained priors closely, then go for the pre-trained weights. If it does not match any, then use the pre-trained weights trained across all priors. In case you want to fine-tune, start from the weights trained across all priors. For optimal performance (if you have enough data) full retraining is generally the most beneficial.
- This really depends on how much data you have per participant and also what parts you are using for inference. If you want to run inference on a held-out participant then train across all subjects. If you want to run on the same participant then e.g. train on the first 50 minutes (or the parts where you have eye-tracking data available) and do inference on the last 10. 
- It matters in all cases where you train across datasets (or fine-tune our weights). We normalize all datasets to be in visual degrees to be able to run the model across all participants and datasets. In case you are running it for the same dataset, then pixel space is fine (you might need to adjust the loss weights to adapt to the larger values)

**Comment by Naubody** [comment]
Agreed with @CYHSM! Will still copy-paste my (slightly adjusted) answer from our email exchange below for others:

1) The minimum preprocessing step is realignment, all others are optional, and at least in my hands, they did not have a massive effect on model performance. I typically do not perform slice time correction, but do not have strong evidence that it would be a problem. The sub-TR resolution does not use reslicing or temporal interpolation, just decoding of more labels.

2a) For your project specifically, I would try fine-tuning one of our models (e.g., using these weights: datasets_1to5.h5). If you fine-tune, your data needs to be in DVA. If you train from scratch, the units do not matter, as you would simply train the model on decoding in pixels. [...] And one more tip, it might depend on the dataset, but for me, fine-tuning worked better with more steps per epoch and fewer epochs (compared to the reverse).

2b) Both the within-participant and across-participant models in the paper are trained using all participants. The difference is that the within-participant model ALSO sees (held-out) data of the participants in the test set in addition to data from other participants, whereas the across-participant model is trained exclusively on data of other participants. The within-participant model tended to be a bit better across the board, but the difference is small, and I typically use the across-participant model.

**Comment by Naubody** [comment]
Hi Maggie, did these answers help? If so, we could close this issue. If not, feel free to reach out again.

**Comment by CYHSM** [comment]
Closing due to inactivity, feel free to repoen @maggiemae13 or create a new issue, thanks.
